src/Backends/Cubics/UNIFAC.cpp

```diff
@@ -195,7 +195,9 @@ double UNIFAC::UNIFACMixture::ln_gamma_R(const double tau, std::size_t i, std::s
     }
 }
 void UNIFAC::UNIFACMixture::activity_coefficients(double tau, const std::vector<double> &z, std::vector<double> &gamma){
-    std::size_t N = z.size();
+    if (this->N != z.size()) {
```

Comment:
Yes, [here](https://github.com/CoolProp/CoolProp/blob/master/src/Backends/Cubics/UNIFAC.cpp#L247).
The one I removed is because I do not want it to interfere with `set_components`, which is the place it should be set (ie. I corrected something I forgot to change earlier).